apis/eventing/v1alpha1/cloudeventsource_types.go

```diff
@@ -93,14 +126,67 @@ type EventSubscription struct {
 }
 
 func init() {
-	SchemeBuilder.Register(&CloudEventSource{}, &CloudEventSourceList{})
+	SchemeBuilder.Register(&CloudEventSource{}, &CloudEventSourceList{}, &ClusterCloudEventSource{}, &ClusterCloudEventSourceList{})
+}
+
+func (ces *CloudEventSource) GetKind() string {
+	return ces.Kind
+}
+
+func (ces *CloudEventSource) GetName() string {
+	return ces.Name
+}
+
+func (ces *CloudEventSource) GetNamespace() string {
+	return ces.Namespace
+}
+
+func (ces *CloudEventSource) GetSpec() CloudEventSourceSpec {
+	return ces.Spec
+}
+
+func (ces *CloudEventSource) GetStatus() CloudEventSourceStatus {
+	return *ces.Status.DeepCopy()
+}
+
+func (ces *CloudEventSource) GetGeneration() int64 {
+	return ces.Generation
 }
 
 // GenerateIdentifier returns identifier for the object in for "kind.namespace.name"
 func (ces *CloudEventSource) GenerateIdentifier() string {
 	return v1alpha1.GenerateIdentifier("CloudEventSource", ces.Namespace, ces.Name)
 }
 
+func (cces *ClusterCloudEventSource) GetKind() string {
+	return cces.Kind
+}
+
+func (cces *ClusterCloudEventSource) GetName() string {
+	return cces.Name
+}
+
+func (cces *ClusterCloudEventSource) GetNamespace() string {
+	return cces.Namespace
+}
+
+func (cces *ClusterCloudEventSource) GetSpec() CloudEventSourceSpec {
+	return cces.Spec
+}
+
+func (cces *ClusterCloudEventSource) GetStatus() CloudEventSourceStatus {
+	return *cces.Status.DeepCopy()
+}
+
+func (cces *ClusterCloudEventSource) GetGeneration() int64 {
+	return cces.Generation
+}
```

Comment:
which would also make many of these not necessary
```suggestion
func (cces *ClusterCloudEventSource) GetSpec() CloudEventSourceSpec {
	return cces.Spec
}

func (cces *ClusterCloudEventSource) GetStatus() CloudEventSourceStatus {
	return *cces.Status.DeepCopy()
}
```